tests/python/gaia-ui-tests/gaiatest/apps/wallpaper/app.py

```diff
@@ -17,7 +17,6 @@ def tap_wallpaper_by_index(self, index):
         Wait(self.marionette).until(
             lambda m: len(m.find_elements(*self._stock_wallpapers_locator)) >= (index + 1),
             message='%d wallpaper(s) not present after timeout' % (index + 1))
-        self.tap_element_from_system_app(
-            self.marionette.find_elements(*self._stock_wallpapers_locator)[index])
-        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name != self.name)
+        self.marionette.find_elements(*self._stock_wallpapers_locator)[index].tap()
```

Comment:
YEa I changed it in the wrong way I think, It should be like this:
self.tap_element_from_system_app(
            self.marionette.find_elements(*self._stock_wallpapers_locator)[index])
        self.apps.wait_to_not_be_displayed()